proposals/conversion-webhook.md

```diff
@@ -0,0 +1,73 @@
+
+Kyverno policy fields can be classified into three categories. The first category includes fields that have been deprecated and replaced with alternative options. The second category includes fields that have been deprecated with no available alternatives. The third category includes fields that have been moved to their respective rules from the `spec` level.
+
+1. Deprecated fields that are replaced with alternative options can be easily converted from one version to another. For example, `ClusterPolicy` v1 that uses `spec.generateExistingOnPolicyUpdate` can be converted to `spec.generateExisting` in v2 using the conversion webhook. There's no loss of information.
+
+2. Deprecated fields with no available alternatives such as `spec.schemaValidation` must now be stored as an annotation in `v2`. The annotation `policies.kyverno.io/conversion-data` can be used to store the source object as json data in the destination object. Alternatively, each deprecated field can have a specific annotation. For example, in the case of `spec.schemaValidation`, it will be `policies.kyverno.io/schema-validation`.
+
+3. Fields that are at the `spec` level and related to specific rules like:
+   - `spec.mutateExistingOnPolicyUpdate`
+   - `spec.generateExisting`
+   - `spec.validationFailureAction`
+   - `spec.validationFailureActionOverrides`
+
+   will be moved to their respective rules and can be easily converted from one version to another using the conversion webhook. There's no loss of information.
+
+As per the Kubernetes migration guide, the initial plan for API migration to `v2` will be as follows:
+
+1. For Kyverno v1.12:
+   - Migrate v2beta1 -> v2beta2.
+   - Deprecate v2beta1.
+   - The storage version is v1.
+2. For Kyverno v1.13:
+   - Migrate v2beta2 -> v2.
+   - Deprecate v2beta2 and v1.
+   - The storage version is v1.
+3. For Kyverno 1.14:
+   - The storage version will be v2.
+4. For Kyverno 1.15:
+   - v2beta1 will be removed.
+5. For Kyverno 1.16:
+   - v2beta2 will be removed.
+
+But introducing a new beta version `v2beta2` could complicate the implementation of the conversion webhook since each version must be converted to/from the storage version. An alternative solution is to migrate to `v2` directly as follows:
+1. For Kyverno v1.12:
+   - Migrate v2beta1 -> v2.
+   - Deprecate v2beta1 and v1.
+   - The storage version is v1.
+2. For Kyverno v1.13:
+   - The storage version will be v2.
+3. For Kyverno 1.14:
+   - v2beta1 will be removed.
+
+Note that multiple versions may exist in storage if they were written before the storage version changes – changing the storage version only affects how objects are created/updated after the change.
+
+## Implementation
```

Comment:
I think implementation is the most important thing to clarify here. The two standouts to me are:
1. Testing of the conversion
2. Deployment of the CRDs with the webhook included in helm (size limitation issues + templating necessity)

I think the proposal here might change slightly when implementation plans become more clear.